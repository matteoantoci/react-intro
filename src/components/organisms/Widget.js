import React from 'react';
import {useStore} from "../../store/store";
import {counterButtonClick} from "../../store/actions";
import {Counter} from "../molecules/Counter";

export const Widget = () => {
  const { state, dispatch } = useStore();
  const { count } = state;
  const handleIncrement = () => dispatch(counterButtonClick());
  return (
    <div className="App-Widget mb16">
      <div className="mb16">Here is my widget!</div>
      <Counter count={count} onClick={handleIncrement} />
      <div>{`I can see widget state too: ${count}`}</div>
    </div>
  );
};
