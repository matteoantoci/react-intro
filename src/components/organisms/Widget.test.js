import React from "react";
import { Widget } from './Widget'
import renderer from 'react-test-renderer';

describe('Widget', () => {
  it('renders correctly', () => {
    const count = 999;
    const fakeCallback = () => {};

    const tree = renderer
      .create(<Widget handleIncrement={fakeCallback} count={count} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
