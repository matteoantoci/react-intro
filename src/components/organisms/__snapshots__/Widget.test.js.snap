// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget renders correctly 1`] = `
<div
  className="App-Widget mb16"
>
  <div
    className="mb16"
  >
    Here is my widget!
  </div>
  <div
    className="App-Counter mb16"
  >
    <div>
      This is a counter molecule
    </div>
    <button
      onClick={[Function]}
    >
      User clicked 999 times
    </button>
  </div>
  <div>
    I can see widget state too: 999
  </div>
</div>
`;
